src/main/java/com/bgsoftware/superiorskyblock/island/IslandUtils.java

```diff
@@ -218,7 +218,8 @@ public static void handleKickPlayer(SuperiorPlayer caller, Island island, Superi
     }
 
     public static void handleKickPlayer(SuperiorPlayer caller, String callerName, Island island, SuperiorPlayer target) {
-        plugin.getEventsBus().callIslandKickEvent(caller, target, island);
+        if(plugin.getEventsBus().callIslandKickEvent(caller, target, island))
```

Comment:
callIslandKickEvent will return whether the event was successfully called.
Therefore, you want to negate the check - it should return when the method returns false.